sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/PushDownLeftSemiAntiJoin.scala

```diff
@@ -61,9 +61,10 @@ object PushDownLeftSemiAntiJoin extends Rule[LogicalPlan]
       }
 
     // LeftSemi/LeftAnti over Aggregate, only push down if join can be planned as broadcast join.
-    case join @ Join(agg: Aggregate, rightOp, LeftSemiOrAnti(_), _, _)
+    case join @ Join(agg: Aggregate, rightOp, LeftSemiOrAnti(_), joinCond, _)
         if agg.aggregateExpressions.forall(_.deterministic) && agg.groupingExpressions.nonEmpty &&
           !agg.aggregateExpressions.exists(ScalarSubquery.hasCorrelatedScalarSubquery) &&
+          canPushThroughCondition(agg.children, joinCond, rightOp) &&
```

Comment:
I think we should rewrite the `joinCondition` assuming the join has already been pushed through Aggregate. That said, we need to do alias replacement for `joinCondition` first. cc @EnricoMi

Reply:
I don't understand. The `canPushThroughCondition` is called before the `Join` is being pushed through the `Aggregate`, it has been added to prevent this from happening in this situation. The other cases (e.g. `Union`) are calling into `canPushThroughCondition` equivalently.

Reply:
This is probably an existing issue. If the `Aggregate` does `a as x, b as y`, then the join condition won't reference `a` and `b` and this check is useless.

Reply:
nvm, `canPushThroughCondition` checks the right side references of the join condition, and check if the right side references have conflict expr ID with left side plan (below Project) output. It doesn't care about the left side references of the join condition.